storage: TestMVCCHistories failed

storage.TestMVCCHistories [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/14127611?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/14127611?buildTab=artifacts#/) on master @ [d4306a9e895eaf50ef9b87b4ba09581447f7dd70](https://github.com/cockroachdb/cockroach/commits/d4306a9e895eaf50ef9b87b4ba09581447f7dd70):


```
        >> at end:
        data: "k1"/50.000000000,0 -> /INT/1
        data: "k1"/40.000000000,0 -> {localTs=30.000000000,0}/<empty>
        data: "k1"/30.000000000,0 -> /BYTES/v
        data: "k1"/20.000000000,0 -> {localTs=10.000000000,0}/BYTES/v
        data: "k2"/50.000000000,0 -> /INT/1
        data: "k2"/40.000000000,0 -> /<empty>
        data: "k2"/30.000000000,0 -> /BYTES/v
        data: "k2"/20.000000000,0 -> /BYTES/v
        data: "k3"/50.000000000,0 -> {localTs=40.000000000,0}/INT/1
        data: "k3"/40.000000000,0 -> /<empty>
        data: "k3"/30.000000000,0 -> {localTs=20.000000000,0}/BYTES/v
        data: "k3"/20.000000000,0 -> /BYTES/v
        stats: key_count=3 key_bytes=153 val_count=12 val_bytes=107 live_count=3 live_bytes=77 gc_bytes_age=11650
    datadriven.go:144: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/12623/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/storage/storage_test_/storage_test.runfiles/com_github_cockroachdb_cockroach/pkg/storage/testdata/mvcc_histories/local_timestamp:99:
        run [2 args]
        with ts=60
          del_range localTs=50 k=k1 end=k2
          del_range localTs=60 k=k2 end=k3
          del_range localTs=70 k=k3 end=k4
        ----
        >> del_range localTs=50 k=k1 end=k2 ts=60
        del_range: "k1"-"k2" -> deleted 1 key(s)
        stats: key_bytes=+12 val_count=+1 val_bytes=+14 live_count=-1 live_bytes=-21 gc_bytes_age=+1880
        >> del_range localTs=60 k=k2 end=k3 ts=60
        del_range: "k2"-"k3" -> deleted 1 key(s)
        stats: key_bytes=+12 val_count=+1 live_count=-1 live_bytes=-21 gc_bytes_age=+1320
        >> del_range localTs=70 k=k3 end=k4 ts=60
        del_range: "k3"-"k4" -> deleted 1 key(s)
        stats: key_bytes=+12 val_count=+1 live_count=-1 live_bytes=-35 gc_bytes_age=+1880
        >> at end:
        data: "k1"/60.000000000,0 -> {localTs=50.000000000,0}/<empty>
        data: "k1"/50.000000000,0 -> /INT/1
        data: "k1"/40.000000000,0 -> {localTs=30.000000000,0}/<empty>
        data: "k1"/30.000000000,0 -> /BYTES/v
        data: "k1"/20.000000000,0 -> {localTs=10.000000000,0}/BYTES/v
        data: "k2"/60.000000000,0 -> /<empty>
        data: "k2"/50.000000000,0 -> /INT/1
        data: "k2"/40.000000000,0 -> /<empty>
        data: "k2"/30.000000000,0 -> /BYTES/v
        data: "k2"/20.000000000,0 -> /BYTES/v
        data: "k3"/60.000000000,0 -> /<empty>
        data: "k3"/50.000000000,0 -> {localTs=40.000000000,0}/INT/1
        data: "k3"/40.000000000,0 -> /<empty>
        data: "k3"/30.000000000,0 -> {localTs=20.000000000,0}/BYTES/v
        data: "k3"/20.000000000,0 -> /BYTES/v
        stats: key_count=3 key_bytes=189 val_count=15 val_bytes=121 gc_bytes_age=16730
    mvcc_history_test.go:2594: unknown reader type "readonly"
    --- FAIL: TestMVCCHistories/local_timestamp (0.03s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/storage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMVCCHistories.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

storage.TestMVCCHistories [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/14133467?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/14133467?buildTab=artifacts#/) on master @ [ee90e6426da817f429799690a0a310829f94c678](https://github.com/cockroachdb/cockroach/commits/ee90e6426da817f429799690a0a310829f94c678):


```
        put k=i ts=3 v=i3
        put k=j ts=3 v=j3
        put k=k ts=3 v=k3
        put k=l ts=3 v=l3
        put k=m ts=3 v=m3
        put k=n ts=3 v=n3
        put k=o ts=3 v=o3
        put k=p ts=3 v=p3
        put k=q ts=3 v=q3
        put k=r ts=3 v=r3
        put k=k ts=4 v=k4
        put k=l ts=4 v=l4
        put k=m ts=4 v=m4
        put k=n ts=4 v=n4
        put_rangekey k=c end=g ts=1
        put_rangekey k=c end=p ts=3
        put_rangekey k=m end=p ts=5
        ----
        >> at end:
        rangekey: {c-g}/[3.000000000,0=/<empty> 1.000000000,0=/<empty>]
        rangekey: {g-m}/[3.000000000,0=/<empty>]
        rangekey: {m-p}/[5.000000000,0=/<empty> 3.000000000,0=/<empty>]
        data: "a"/3.000000000,0 -> /BYTES/a3
        data: "b"/3.000000000,0 -> /BYTES/b3
        data: "c"/3.000000000,0 -> /BYTES/c3
        data: "d"/3.000000000,0 -> /BYTES/d3
        data: "e"/3.000000000,0 -> /BYTES/e3
        data: "e"/2.000000000,0 -> /BYTES/e2
        data: "f"/3.000000000,0 -> /BYTES/f3
        data: "f"/2.000000000,0 -> /BYTES/f2
        data: "g"/3.000000000,0 -> /BYTES/g3
        data: "g"/2.000000000,0 -> /BYTES/g2
        data: "h"/3.000000000,0 -> /BYTES/h3
        data: "h"/2.000000000,0 -> /BYTES/h2
        data: "i"/3.000000000,0 -> /BYTES/i3
        data: "j"/3.000000000,0 -> /BYTES/j3
        data: "k"/4.000000000,0 -> /BYTES/k4
        data: "k"/3.000000000,0 -> /BYTES/k3
        data: "l"/4.000000000,0 -> /BYTES/l4
        data: "l"/3.000000000,0 -> /BYTES/l3
        data: "m"/4.000000000,0 -> /BYTES/m4
        data: "m"/3.000000000,0 -> /BYTES/m3
        data: "n"/4.000000000,0 -> /BYTES/n4
        data: "n"/3.000000000,0 -> /BYTES/n3
        data: "o"/3.000000000,0 -> /BYTES/o3
        data: "p"/3.000000000,0 -> /BYTES/p3
        data: "q"/3.000000000,0 -> /BYTES/q3
        data: "r"/3.000000000,0 -> /BYTES/r3
    mvcc_history_test.go:2594: unknown reader type "readonly"
    --- FAIL: TestMVCCHistories/range_tombstone_gets_ts_conflict (0.08s)
```

Parameters:
 - <code>TAGS=bazel,gss</code>
 - <code>stress=true</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMVCCHistories.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/storage/mvcc_history_test.go
+++ b/pkg/storage/mvcc_history_test.go
@@ -2579,10 +2579,10 @@ func (e *evalCtx) getTxn(opt optArg) *roachpb.Transaction {
 // newReader returns a new (metamorphic) reader for use by a single command. The
 // caller must call Close on the reader when done.
 func (e *evalCtx) newReader() storage.Reader {
-	switch mvccHistoriesReader {
+	switch strings.ToLower(mvccHistoriesReader) {
 	case "engine":
 		return noopCloseReader{e.engine}
-	case "reader", "readOnly":
+	case "reader", "readonly":
 		return e.engine.NewReader(storage.StandardDurability)
 	case "batch":
 		return e.engine.NewBatch()

